src/gui/FileBrowser.cpp

```diff
@@ -624,42 +625,57 @@ void FileBrowserTreeWidget::focusOutEvent(QFocusEvent* fe)
 	QTreeWidget::focusOutEvent(fe);
 }
 
+void FileBrowserTreeWidget::contextMenuEvent(QContextMenuEvent* e)
+{
+#ifdef LMMS_BUILD_APPLE
+	QString fileManager = tr("Finder");
+#elif defined(LMMS_BUILD_WIN32)
+	QString fileManager = tr("Explorer");
+#else
+	QString fileManager = tr("file manager");
+#endif
 
+	QTreeWidgetItem* item = itemAt(e->pos());
 
+	auto file = dynamic_cast<FileItem*>(item);
 
-void FileBrowserTreeWidget::contextMenuEvent(QContextMenuEvent * e )
-{
-	auto file = dynamic_cast<FileItem*>(itemAt(e->pos()));
-	if( file != nullptr && file->isTrack() )
+	QMenu contextMenu(this);
+
+	auto dir = dynamic_cast<Directory*>(item); // TODO: this might not be a great way to check if it's a directory
```

Comment:
Using `dynamic_cast` here IMO is fine, though maybe we could improve the interface a bit for these classes. Maybe you were thinking of using `isDirectory` and `isFile` `virtual` functions? In which case I would agree with you, but any kind of refactoring here can be done later.

Reply:
I've already recommend a full snippet example... `QTreeWidgetItem` stores its time in an enum for this exact purpose.

Reply:
Link to enum approach: https://github.com/LMMS/lmms/pull/7700#discussion_r1957202779

Reply:
> Link to enum approach: https://github.com/LMMS/lmms/pull/7700#discussion_r1957202779

Though this approach still ends up doing the `dynamic_cast` anyways. Guidelines state that [virtual functions should be used preferred instead of casting](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-use-virtual), which means that we would need a better abstract interface that would work for both `FileItem` and `Directory`.

Reply:
The new approach casts on a predictable, well-known type, it no longer relies on casting as the mechanism for detecting the type directly addressing the "better way to do this comment".  Combining these two related types into a common interface is out of scope for the OP.